builtin/joined_dataset.cc

```diff
@@ -398,12 +399,20 @@ struct JoinedDataset::Itl
                     //return all rows
                     sideCondition = SqlExpression::TRUE;
 
+                    shared_ptr<SqlExpression> sideWhere;
```

Comment:
Can you add a comment here stating why we need this extra step?  This code is getting harder to understand everyday! :-(